glusterd2/volgen/store-utils.go

```diff
@@ -48,3 +48,35 @@ func GetVolfile(volfileID string) ([]byte, error) {
 	}
 	return resp.Kvs[0].Value, nil
 }
+
+// DeleteVolumeVolfiles deletes all the Volfiles belonging to a Volume
+func DeleteVolumeVolfiles(volname string) error {
+	resp, err := store.Get(context.TODO(), volfilePrefix+volname, clientv3.WithPrefix())
```

Comment:
You could directly do a `store.Delete(ctx, prefix, clientv3.WithPrefix())` to delete everything under the prefix. No need to fetch each entry individually. Same for the snaps volfiles below.

Tie both together using etcd transaction to ensure all happen with a single call.
```go
resp, err = store.Store.Txn(ctx).If().
  Then(
    clientv3.OpDel(volprefix, clientvt.WithPrefix()),
    clientv3.OpDel(volpreifx),
    clientv3.OpDel(snapPrefix, clientv3.WithPrefix()),
    clientv3.OpDel(snapPrefix)
  ).Else().Commit()
```

This is what I believe @prashanthpai commented about before.

Reply:
good one. thanks for the suggestion.